lib/u3d_core/update_checker/update_checker.rb

```diff
@@ -0,0 +1,135 @@
+
+    def self.show_update_status(gem_name, current_version)
+      wait_for_results unless update_available?(gem_name, current_version)
+      show_update_message(gem_name, current_version) if update_available?(gem_name, current_version)
+    end
+
+    # Show a message to the user to update to a new version of u3d
+    # Use this method, as this will detect the current Ruby environment and show an
+    # appropriate message to the user
+    def self.show_update_message(gem_name, current_version)
+      available = server_results[gem_name]
+      puts("")
+      puts('#######################################################################')
+      if available
+        puts("# #{gem_name} #{available} is available. You are on #{current_version}.")
+      else
+        puts("# An update for #{gem_name} is available. You are on #{current_version}.")
+      end
+      puts("# You should use the latest version.")
+      puts("# Please update using `#{update_command(gem_name: gem_name)}`.")
+
+      # this could be fetched from the gem
+      puts("# To see what's new, open https://github.com/DragonBox/#{gem_name}/releases.") if U3dCore::Env.truthy?("U3D_HIDE_CHANGELOG")
+
+      if !Helper.bundler? && Random.rand(5) == 1
+        # We want to show this message from time to time, if the user doesn't use bundler, nor bundled fastlane
+        puts('#######################################################################')
+        puts("# Run `sudo gem cleanup` from time to time to speed up fastlane")
+      end
+      puts('#######################################################################')
+      Changelog.show_changes(gem_name, current_version, update_gem_command: UpdateChecker.update_command(gem_name: gem_name)) unless U3dCore::Env.truthy?("U3D_HIDE_CHANGELOG")
+
+      ensure_rubygems_source
+    end
+
+    # The command that the user should use to update their mac
+    def self.update_command(gem_name: "u3d")
+      if Helper.bundler?
+        "bundle update #{gem_name.downcase}"
+      # elsif Helper.homebrew?
+      #  "fastlane update_fastlane"
+      # elsif Helper.mac_app?
+      #  "the Fabric app. Launch the app and navigate to the fastlane tab to get the most recent version."
+      else
+        "sudo gem install #{gem_name.downcase}"
+      end
+    end
+
+    # Check if RubyGems is set as a gem source
+    # on some machines that might not be the case
+    # and then users can't find the update when
+    # running the specified command
+    def self.ensure_rubygems_source
+      return if `gem sources`.include?("https://rubygems.org")
+      puts("")
+      UI.error("RubyGems is not listed as your Gem source")
+      UI.error("You can run `gem sources` to see all your sources")
+      UI.error("Please run the following command to fix this:")
+      UI.command("gem sources --add https://rubygems.org")
+    end
+
+    def self.fetch_latest(gem_name)
+      JSON.parse(Excon.get(generate_fetch_url(gem_name)).body)["version"]
```

Comment:
Do we need to use `Excon`? We already make a number of HTTP calls and have helper for this kind of operation. I'd rather we don't multiply the way to perform this type of operation